qemu/tests/uefishell.py

```diff
@@ -103,13 +102,13 @@ def var_copy(self):
         since the operations inside uefishell will be recorded
         into VAR file and may affact later test cases.
         """
-        source_var_path = "/usr/share/OVMF/OVMF_VARS.fd"
-        path = storage.get_image_filename_filesytem(
-            self.params, data_dir.get_data_dir())
-        vars_path = "%s.fd" % path
+        source_var_path = self.params["ovmf_path"]
+        source_var_path += ('/' + self.params["ovmf_vars_filename"])
+        dev = self.vm.devices.get_by_params({"node-name": "file_ovmf_vars"})[0]
```

Comment:
```suggestion
        dev = self.vm.devices.get_by_properties({"aobject": "ovmf_vars"})[0]
        if vm.check_capability(Flags.BLOCKDEV):
            file_ovmf_vars = dev.get_param('filename')
        else:
            file_ovmf_vars = dev.get_param('file')
```

Reply:
Thanks for your guidance,the code has been updated.